### PR TITLE
feat(api): Add initial annotations api module and factory

### PR DESCRIPTION
**Pull request by jstoffan**
Todo
- [x] Improve types for API method parameters
- [x] Resolve all TODOs once endpoints are updated

### CODE REVIEW

File: src/api/APIFactory.js [hunk 1]
```diff
@@ -592,6 +603,20 @@ class APIFactory {
         this.boxEditAPI = new BoxEditAPI();
         return this.boxEditAPI;
     }
+
+    /**
+     * API for Annotations
+     *
+     * @return {AnnotationsAPI} AnnotationsAPI instance
+     */
+    getAnnotationsAPI(shouldDestroy: boolean): AnnotationsAPI {
```
Comment (mxiao6) [inline src/api/APIFactory.js:612]: Is it a good idea to add a default value?

Comment (jstoffan) [reply src/api/APIFactory.js:612]: Yeah, good idea.

File: src/api/Annotations.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        });
+    }
+
+    getAnnotations(
+        fileId: string,
+        fileVersionId?: string,
+        successCallback: (annotations: Annotation[]) => void,
+        errorCallback: (e: ElementsXhrError, code: string) => void,
+        limit?: number,
+        shouldFetchAll?: boolean,
```
Comment (mickr) [inline src/api/Annotations.js:80]: Default for this as well?

File: src/common/types/annotations.js [hunk 3]
```diff
@@ -1,33 +1,57 @@
...
+export type TargetRegion = {
     location: Page,
     shape?: Rect,
     type: 'region',
 };
 
-type Target = TargetRegion;
+export type Target = TargetRegion;
+
+export type Annotation = {
```
Comment (mickr) [inline src/common/types/annotations.js:34]: 👍

File: src/api/Annotations.js (hunk 2 above)
Comment (jstoffan) [reply src/api/Annotations.js:80]: Already has one in the underlying method.

**Pull request merged**